.github/workflows/docker.yml

```diff
@@ -0,0 +1,47 @@
+name: Build Docker images 
+
+on:
+  push: 
+    branches:
+      - master
+    tags:
+      - '*'
+
+jobs: 
+  build:
+    name: Build Docker image 
+    runs-on: ubuntu-latest
+    environment: scipy-dev
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+      - name: Lint Docker 
+        uses: brpaz/hadolint-action@v1.2.1
+        with: 
+          dockerfile: ./tools/Docker-dev/Dockerfile
+      - name: Get refs
+        shell: bash
+        run: |
+          export raw_branch=${GITHUB_REF#refs/heads/}
+          echo "::set-output name=branch::${raw_branch//\//-}"
+          echo "::set-output name=date::$(date +'%Y%m%d')"
+          echo "::set-output name=sha8::$(echo ${GITHUB_SHA} | cut -c1-8)"
+        id: getrefs
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build and push
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          context: "./tools/Docker-dev"
+          file: "./tools/Docker-dev/Dockerfile"
+          push: ${{ github.event_name != 'pull_request' }}
```

Comment:
Probably also needs a condition that the branch is master. There's no need for building on maintenance branches.

Reply:
At the moment this action should only run on pushes to master and new tags. From above:

```
on:
  push: 
    branches:
      - master
    tags:
      - '*'
   ```

We can add other conditions as needed